### ISSUE TITLE
You can Tween Enums

### ISSUE THREAD

**Issue by ModernBalloonie** [Yellow-Dog-Man/Resonite-Issues#442]
### Describe the bug?

Probably not intended, I would assume that is, but you can make TweenValues of Enums, which spawns a Tween component which disables itself probably because of an exception.

### To Reproduce

Spawn a TweenValue node of any Enum, then make a valuefield of that enum and try to tween it, This ends up happening

https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/82356528/c345806f-c500-4817-9028-8b6fbcd2adea

### Expected behavior

Well, i'm not sure if this is intended, so i'm not sure what I expect to happen.

### Resonite Version Number

2023.10.20.831

### What Platforms does this occur on?

Windows

### What headset if any do you use?

Valve Index

### Log Files

[DESKTOP-QBGCSAQ - 2023.10.20.831 - 2023-10-25 17_17_07.log](https://github.com/Yellow-Dog-Man/Resonite-Issues/files/13171780/DESKTOP-QBGCSAQ.-.2023.10.20.831.-.2023-10-25.17_17_07.log)

### Reporters

modernballoonie on Discord

**Comment by ProbablePrime** [comment]
As there's no downside(crashes, unexpected behavior) etc, I don't see an issue.

This works because enums are numbers under the hood. I dont think we need to explicitly prevent this and can even think of a few educational uses, for example showing material enums via animations etc.

**Comment by ProbablePrime** [comment]
Oh the component exception does need to be handled though woops.

**Comment by ModernBalloonie** [comment]
Yup, the tween component just doesn't want to exist. Maybe I should of brought more focus to that instead of the ability to tween enums.

**Comment by Frooxius** [comment]
@ProbablePrime We should not handle the exception, we need to make sure it's not thrown in the first place.

We need to disallow creating those generic types with enums. We do the same for other nodes, but it's not implemented here.

Or specifically, we only validate types that support lerping, which is done in generic way through Coder<T>.

**Comment by ProbablePrime** [comment]
@Frooxius understood. When i said handled i meant "resolved" not "handled" as in caught.

**Comment by Redd56** [comment]
to add and point out a sillything, you can tween a bool if you so please
